cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -693,6 +723,36 @@ func instanceStatusFromProvisioningState(provisioningState *string) *cloudprovid
 	return status
 }
 
+func powerStateFromStatuses(statuses []compute.InstanceViewStatus) string {
+	for _, status := range statuses {
+		if status.Code == nil || !isKnownPowerState(*status.Code) {
+			continue
+		}
+		return *status.Code
+	}
+
+	// PowerState is not set if the VM is still creating (or has failed creation),
+	// so the absence of a PowerState is treated the same as a VM that is stopped
+	return PowerStateStopped
```

Comment:
I think we should let `PowerStateUnknown` be the default in this case. The assumption in the comment could change anytime from the updates on their side. 
And since this case is not having a behavior difference between returning `PowerStateUnknown` and `PowerStateStopped` for now, the earlier one would be safer.

Reply:
Agreed, I've updated to make `vmPowerStateUnknown` the default case -- thanks!